events/guild_events.go

```diff
@@ -21,7 +21,8 @@ type GuildUpdate struct {
 
 // GuildAvailable is called when an unavailable discord.Guild becomes available
 type GuildAvailable struct {
-	*GenericGuild
+	*GenericEvent
```

Comment:
I think I would prefer it if you removed the `Guild` field from the `events.GenericGuild` andf then just add it to all the other events